trk/src/Screens/TabScreens/Record/Backend/logic.js

```diff
@@ -26,6 +28,42 @@ export const useHomeScreenLogic = (props) => {
 
     const dispatch = useDispatch();
 
+    //Next PR these will actually be calculated rather than hardcoded (set the true value to the option with the most messages in JSON file)
+    let isFirstTimeUser = false;
+    let isFTUNotFirstClimb = false;
+    let isReturningUserFirstClimbOfDay = false;
+    let isReturningUserFirstClimbOfAnotherSession = true;
+    let isReturningUserNotFirstClimb = false;
+
+
+    const [selectedMessage, setSelectedMessage] = useState('');
+    
+
+    function randomIndex(array) {
+        return Math.floor(Math.random() * array.length);
+    }
+
+    const selectRandomMessage = useCallback(() => {
+        let message;
+        if (isFirstTimeUser) {
+            message = tapMessage.FTUFirstClimb[randomIndex(tapMessage.FTUFirstClimb)];
+        } else if (isFTUNotFirstClimb) {
+            message = tapMessage.FTUNotFirstClimb[randomIndex(tapMessage.FTUNotFirstClimb)];
+        } else if (isReturningUserFirstClimbOfDay) {
+            message = tapMessage.ReturningUserFirstClimbOfDay[randomIndex(tapMessage.ReturningUserFirstClimbOfDay)];
+        } else if (isReturningUserFirstClimbOfAnotherSession) {
+            message = tapMessage.ReturningUserFirstClimbOfAnotherSession[randomIndex(tapMessage.ReturningUserFirstClimbOfAnotherSession)];
+        } else if (isReturningUserNotFirstClimb) {
+            message = tapMessage.ReturningUserNotFirstClimb[randomIndex(tapMessage.ReturningUserNotFirstClimb)];
+        } else {
+            // Default message or other logic
+            message = "Completed a climb? Tap below to save your achievement";
+        }
+        setSelectedMessage(message);
+    }, [isFirstTimeUser, isFTUNotFirstClimb, isReturningUserFirstClimbOfDay]);
+
+    useFocusEffect(selectRandomMessage);
+
```

Comment:
line 65: every time you navigate out and back to record, the randomMessage is re-selected